app/models/cart.rb

```diff
@@ -32,4 +37,18 @@ def total
     end
   end
 
+  def add_quantity(quantity, item, inventory)
+    if @contents.values.first < inventory
+      quantity.times do
+        add_item(item.id.to_s)
+      end
+    end
+  end
+
+  def remove_quantity(quantity, item)
+    quantity.times do
+      remove_item(item.id.to_s)
+    end
```

Comment:
I created a loop to either add or subtract quantity based on user input. quantity is passed in through the controller and that determines out many times the loop runs.